src/Core/Foundation/Version/Version.php

```diff
@@ -0,0 +1,223 @@
+     * Returns if the current PrestaShop version is less than the provided version
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    public function isLessThan($version)
+    {
+        return $this->versionCompare($version, '<');
+    }
+
+    /**
+     * Returns if the current PrestaShop version is less than or equal to the provided version
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    public function isLessThanOrEqualTo($version)
+    {
+        return $this->versionCompare($version, '<=');
+    }
+
+    /**
+     * Returns if the current PrestaShop version is equal to the provided version
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    public function isEqualTo($version)
+    {
+        return $this->versionCompare($version, '=');
+    }
+
+    /**
+     * Returns if the current PrestaShop version is not equal to the provided version
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    public function isNotEqualTo($version)
+    {
+        return $this->versionCompare($version, '!=');
+    }
+
+    /**
+     * Compares the current PrestaShop version with the provided version depending on the provided operator
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     * @param $operator Operator for version_compare(), allowed values are: <, lt, <=, le, >, gt, >=, ge, ==, =, eq, !=, <>, ne
+     *
+     * @return boolean Result of the comparison
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    private function versionCompare($version, $operator)
+    {
+        try {
+            $this->checkVersion($version);
+        } catch(InvalidVersionException $e) {
+            throw new InvalidArgumentException($e->getMessage());
+        }
+
+        return version_compare($this->version, $version, $operator);
+    }
+
+    /**
+     * Checks if a given version is a valid version string
+     *
+     * @param $version
+     *
+     * @throws InvalidVersionException If the provided version is invalid
+     */
+    private function checkVersion($version)
+    {
+        if (!is_string($version)) {
+            throw new InvalidVersionException('A valid version must be a string.');
+        }
+
+        if (!preg_match('/^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$/', $version)) {
```

Comment:
Watch out, as I said, not all versions have x.x.x.x. Can be x.x, x.x.x, and x.x.x.x (and sometimes, x.x.x-(alpha|beta|rc\d+) )

Reply:
@PierreRambaud I know, I already answered above :)